Automattic/jetpack
# Publicize: allow all admins to toggle the feature on / off

This is an alternative to #34202

## Proposed changes:

*
All admins should be able to toggle features on and off, just like they can under Jetpack > Settings > Modules.

This commit also hides the upgrade link for folks that have not connected their account to WordPress.com yet, since they will not be able to purchase the upgrade without being connected.

**When user is not connected**

<img width="1075" alt="Screenshot 2023-11-20 at 11 33 47" src="https://github.com/Automattic/jetpack/assets/426388/47e0fba4-9d70-4803-abdb-30fbb04cb4a2">

**When user is connected**

<img width="1094" alt="Screenshot 2023-11-20 at 11 33 54" src="https://github.com/Automattic/jetpack/assets/426388/148a57e1-acf2-463a-aa4a-7e5d5e7ad6e3">

### Other information:


## Jetpack product discussion

Related discussion: p1699929025605869-slack-CDLH4C1UZ

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

* Start with a site that's connected to WordPress.com.Start with a site that's connected to WordPress.com.
* Go to Jetpack > Settings > Sharing
* Enable the Publicize feature.
* Go to Users > Add New and create a new admin.
* In a new browser, log in with that user.
* Go to Jetpack > Settings > Sharing. 
    * You should be able to turn the Publicize feature off, but you should not see the upgrade link.
